### ISSUE TITLE
Intermittent Travis "sniff seed" test failure

### ISSUE THREAD

**Issue by bitcoin-solutions** [bitcoin-solutions/multibit-hd#354]
> Failed tests:
>   TrezorWalletTest.testCreateTrezorSoftWalletFromSniffSeed:334
> expected:<'1[8dxk72otf2amyAsjiKnEWhox5CJGQHYGA]'> but
> was:<'1[78nhbLCC4qZgb9YeJ8tkEhbzX6GBwhrBk]'>
> 
> Travis kicked it out on Oracle JDK 8 with this log:
> https://s3.amazonaws.com/archive.travis-ci.org/jobs/47054898/log.txt

**Comment by bitcoin-solutions** [comment]
Just looked through the log - I cannot see anything obvious causing this.
Also the address that does not match is the 5th out of 5 that are given in the test i.e. all of m/44h/0h/0/x where x=0,1,2,3 work fine - only the 4 fails.

Also double checked it works on Mac

**Comment by gary-rowe** [comment]
I'd like to understand why this happened since it's in the key code. 

From the above we see that Travis build [370.3](https://travis-ci.org/bitcoin-solutions/multibit-hd/jobs/47054898) last had the problem, but it first manifested in [368.3](https://travis-ci.org/bitcoin-solutions/multibit-hd/builds/46991751) [log](https://s3.amazonaws.com/archive.travis-ci.org/jobs/46991754/log.txt).

From 371.x onwards we don't have a recurrence. The only difference between 370.x and 371.x was me adding some email recipients to Travis that clearly will have no effect. Travis [shows the difference between 369.x and 370.x](https://github.com/bitcoin-solutions/multibit-hd/compare/6db5949f977e...bd4e8d53216e) contained only updates to the internal help.

In each case it was Oracle 1.8.0_25 that was running (no changes in build environment) so that seems to be the common factor. I've looked over the Oracle release notes and bug library but I can't find anything relating to a crypto library failure.

**Comment by bitcoin-solutions** [comment]
I'll add some more addresses to check with that seed and another seed example to exercise this more.
I think I'll also see if Mike Hearn is regualrly building bitcoinj against Java 8.

**Comment by bitcoin-solutions** [comment]
I've doubled up the number of addresses we are checking for the sniff seed from 5 to 10.
Also the same test is run twice so that we get more feedback on how repeatable it is.

Note also that the address that appears in the test fail is not random it is the one earlier address being checked - from the code:
```
  // m/44h/0h/0h/0/4
  private static final String SNIFF_EXPECTED_ADDRESS_4 = "18dxk72otf2amyAsjiKnEWhox5CJGQHYGA";

  // m/44h/0h/0h/0/5
  private static final String SNIFF_EXPECTED_ADDRESS_5 = "178nhbLCC4qZgb9YeJ8tkEhbzX6GBwhrBk";

**Comment by bitcoin-solutions** [comment]
Whilst not 'solved' I think this issue can be closed now, subject to us keeping an eye on the Travis logs in case it reoccurrs.
Awaiting review and closing.

**Comment by bitcoin-solutions** [comment]
I think I have tracked this down after the recent Travis failure where this occurred.

The temporary directory used in the TrezorWalletTest using the sytem time, to mills, to create the temporary directory name. I think the travis machine is quick enough that it sometimes uses the same dir for two tests. 
I've modified SecureFiles#createTemporaryDirectory so that in addition to the system time it also adds adds a sub directory with a random number (up to Integer.MAX) and returns the sub dir as the temporary directory. This should avoid collisions.

Just checked in (16:69) so we'll see if the build comes back up
